.github/workflows/golangci-lint.yaml

```
@@ -16,8 +16,14 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
+
 jobs:
   golangci:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
```

```suggestion
      contents: read  # for actions/checkout to fetch code
```
do we need this `contents: read` since this is already set on top?

---

I'll remove it, doesn't seem necesary